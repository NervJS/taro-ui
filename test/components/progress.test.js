import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtProgress from '../../.temp/components/activity-indicator/index'

describe('Progress Snap', () => {
  it('render initial Progress', () => {
    const component = renderToString(<AtProgress />)
    expect(component).toMatchSnapshot()
  })

  it('render Progress -- props percent', () => {
    const component = renderToString(<AtProgress percent={25} />)
    expect(component).toMatchSnapshot()
  })

  it('render Progress -- props isHidePercent', () => {
    const component = renderToString(<AtProgress isHidePercent />)
    expect(component).toMatchSnapshot()
  })

  it('render Progress -- props strokeWidth ', () => {
    const component = renderToString(<AtProgress strokeWidth={10} />)
    expect(component).toMatchSnapshot()
  })

  it('render Progress -- props color ', () => {
    const component = renderToString(<AtProgress color='#FF4949' />)
    expect(component).toMatchSnapshot()
  })

  it('render Progress -- props status(progress) ', () => {
    const component = renderToString(<AtProgress status='progress' />)
    expect(component).toMatchSnapshot()
  })

  it('render Progress -- props status(error) ', () => {
    const component = renderToString(<AtProgress status='error' />)
    expect(component).toMatchSnapshot()
  })

  it('render Progress -- props status(success) ', () => {
    const component = renderToString(<AtProgress status='success' />)
    expect(component).toMatchSnapshot()
  })
})
