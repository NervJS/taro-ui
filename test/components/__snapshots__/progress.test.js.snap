// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Snap render Progress -- props color  1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:1px solid #FF4949;border-color:#FF4949 transparent transparent transparent;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:1px solid #FF4949;border-color:#FF4949 transparent transparent transparent;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:1px solid #FF4949;border-color:#FF4949 transparent transparent transparent;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div></div>"`;

exports[`Progress Snap render Progress -- props isHidePercent 1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div></div>"`;

exports[`Progress Snap render Progress -- props percent 1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div></div>"`;

exports[`Progress Snap render Progress -- props status(error)  1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div></div>"`;

exports[`Progress Snap render Progress -- props status(progress)  1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div></div>"`;

exports[`Progress Snap render Progress -- props status(success)  1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div></div>"`;

exports[`Progress Snap render Progress -- props strokeWidth  1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div></div>"`;

exports[`Progress Snap render initial Progress 1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div></div>"`;
